### PR TITLE
Added javadoc for (de)serialize and toByteArray, and added size check…

### PR DESCRIPTION
**Pull request by Siebje**
…s for empty byte arrays

### CODE REVIEW

File: src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java [hunk 1]
```diff
@@ -39,27 +43,34 @@ public int totalSize() {
...
+            try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes)) {
+                DataInputStream dis = new DataInputStream(bais);
+                //Read NE
+                short numberOfEntries = dis.readShort();
+                for (int k = 0; k < numberOfEntries; k++) {
+                    //Read NL
+                    short nameLength = dis.readShort();
+                    byte[] nameBytes = new byte[nameLength];
+                    int read = dis.read(nameBytes, 0, nameLength);
+                    if (read != nameLength) throw new IOException("Error reading parameter name");
```
Comment (johan-riscure) [inline src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java:63]: Can you put {} braces around this, and preferably put the throw on a new line?

File: src/main/java/com/riscure/trs/parameter/traceset/TraceSetParameterMap.java [hunk 2]
```diff
@@ -39,27 +43,34 @@
...
+                //Read NE
+                short numberOfEntries = dis.readShort();
+                for (int k = 0; k < numberOfEntries; k++) {
+                    //Read NL
+                    short nameLength = dis.readShort();
+                    byte[] nameBytes = new byte[nameLength];
+                    int read = dis.read(nameBytes, 0, nameLength);
+                    if (read != nameLength) throw new IOException("Error reading parameter name");
+                    //Read N
+                    String name = new String(nameBytes, StandardCharsets.UTF_8);
```
Comment (johan-riscure) [inline src/main/java/com/riscure/trs/parameter/traceset/TraceSetParameterMap.java:65]: This is essentially the same code as in TraceParameterDefinitionMap. Not sure if it's worth extracting for use by the both of them though.
I would like to see {} braces around the if statement here as well however.

File: src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java (hunk 1 above)
Comment (StefanD986) [reply src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java:63]: I must say I find these one-line `if(...) throw ;` quite a bit more reader friendly than adding brackets around it.

Comment (johan-riscure) [reply src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java:63]: Ok, given that this throws an exception, I'm willing to accept it as is. However, in general I think if-statements without brackets are an abomination.

Comment (Siebje) [reply src/main/java/com/riscure/trs/parameter/trace/definition/TraceParameterDefinitionMap.java:63]: I'm generally a fan of brackets, but I made some exceptions for...well... exceptions :)

File: src/main/java/com/riscure/trs/parameter/traceset/TraceSetParameterMap.java (hunk 2 above)
Comment (Siebje) [reply src/main/java/com/riscure/trs/parameter/traceset/TraceSetParameterMap.java]: extracted to TRSMetaDataUtils (that seemed like the most appropriate place, seeing as how it's interpreting meta data)

**Pull request merged**